### PR TITLE
Add inbox_note_view Tracks Event

### LINKED ISSUE #7735: Update Inbox Tracks
**Description:** currently, we don't have a way to check how many inbox notifications are displayed in the panel. The number of messages can have a negative impact on the inbox effectiveness and we should start tracking the number of messages visible when the user visits the home screen.

**Acceptance criteria:** 

- [ ] Fire an event tracking the number of messages included in the inbox panel when the user visits the inbox. In terms of tracking, we have two options: introduce a new event to track this data point or add a new prop to an existing event: wcadmin_inbox_note_view

### PR DESCRIPTION
**Pull request by adrianduffell**
Fixes #7735

Adds a new tracks event `inbox_panel_view` containing a `total` prop, representing the total amount of notes in the panel. This is fired on every view of the inbox panel (Home Screen and activity menu).

### Detailed test instructions:

- Add the tracks debugger in browser console: `localStorage.setItem( 'debug', 'wc-admin:*' );`
- Load the WooCommerce home screen
- See that a tracks event is made in the browser console, with the `total` value matching the total rendered in the header.

Screen Shot 2021-12-06 at 8 20 10 pm

Screen Shot 2021-12-06 at 8 23 54 pm

### CODE REVIEW

File: client/inbox-panel/test/index.js [hunk 1]
```diff
@@ -1,38 +1,64 @@
...
+
+jest.mock( '@woocommerce/tracks', () => ( { recordEvent: jest.fn() } ) );
 
 const notes = [
 	{
 		id: 1,
 		date_created_gmt: '2019-05-10T16:57:31',
 		is_deleted: false,
 		status: 'unactioned',
+		actions: [],
```
Comment (adrianduffell) [inline client/inbox-panel/test/index.js:33]: These `actions` are expected in a note object when `<InboxPanel />` renders.

File: client/inbox-panel/test/index.js [hunk 2]
```diff
@@ -1,38 +1,64 @@
...
+	const originalModule = jest.requireActual( '@wordpress/data' );
+
+	return {
+		__esModule: true, // Use it when dealing with esModules
+		...originalModule,
+		useSelect: jest.fn().mockReturnValue( {} ),
+	};
+} );
+
+jest.mock( '@woocommerce/tracks', () => ( { recordEvent: jest.fn() } ) );
```
Comment (adrianduffell) [inline client/inbox-panel/test/index.js:25]: I had to add some plumbing as the tests didn't previously render the `<InboxPanel />`

**Review by moon0326 — APPROVED** [review]
Thank you for working on this @adrianduffell 🚀 

It LGTM and tested well 💯

File: client/inbox-panel/test/index.js (hunk 2 above)
Comment (moon0326) [reply client/inbox-panel/test/index.js:25]: Thank you for adding tests 💯

**Pull request merged**